src/main/java/org/opentripplanner/raptor/rangeraptor/transit/AccessEgressFunctions.java

```diff
@@ -47,32 +52,64 @@ public final class AccessEgressFunctions {
-    (l.durationInSeconds() < r.durationInSeconds());
+    (
+      l.stopReachedOnBoard() &&
+      !r.stopReachedOnBoard() ||
+      r.hasOpeningHours() ||
+      l.numberOfRides() < r.numberOfRides() ||
+      l.durationInSeconds() < r.durationInSeconds()
+    );
+
+  /**
+   * Filter Multi-criteria Raptor access and egress paths. This can be used to wash
+   * access/egress paths - paths that are not optimal using this should not be passed into
+   * Raptor - it is a bug.
+   */
+  private static final ParetoComparator<RaptorAccessEgress> MC_COMPARATOR = (l, r) ->
+    (
+      (l.stopReachedOnBoard() && !r.stopReachedOnBoard()) ||
+      r.hasOpeningHours() ||
+      l.numberOfRides() < r.numberOfRides() ||
+      l.durationInSeconds() < r.durationInSeconds() ||
+      l.c1() < r.c1()
+    );
 
   /** private constructor to prevent instantiation of utils class. */
   private AccessEgressFunctions() {}
 
+  /**
+   * Filter non-optimal paths away for the standard search. This method does not
+   * look at the c1 value.
+   */
   static Collection<RaptorAccessEgress> removeNoneOptimalPathsForStandardRaptor(
     Collection<RaptorAccessEgress> paths
   ) {
-    // To avoid too many items in the pareto set we first group the paths by stop,
-    // for each stop we filter it down to the optimal pareto set. We could do this
-    // for multi-criteria as well, but it is likely not so important. The focus for
-    // the mc-set should be that the list of access/egress created in OTP should not
-    // contain to many non-optimal paths.
-    var mapByStop = groupByStop(paths);
-    var set = new ParetoSet<>(STANDARD_COMPARATOR);
-    Collection<RaptorAccessEgress> result = new ArrayList<>();
+    return removeNoneOptimalPaths(paths, STANDARD_COMPARATOR);
+  }
 
-    mapByStop.forEachValue(list -> {
-      set.clear();
-      set.addAll(list);
-      result.addAll(set);
-      return true;
-    });
+  /**
+   * Filter non-optimal paths away for the multi-criteria search. This method should in theory
+   * not remove any paths since the caller should not pass in duplicates, but it turns out that
+   * this happens, so we do it.
+   */
+  static Collection<RaptorAccessEgress> removeNoneOptimalPathsForMcRaptor(
+    Collection<RaptorAccessEgress> paths
+  ) {
+    var result = removeNoneOptimalPaths(paths, MC_COMPARATOR);
+    if (LOG.isDebugEnabled() && result.size() < paths.size()) {
+      var duplicates = new ArrayList<>(paths);
+      duplicates.removeAll(result);
+      // Note! This does not provide enough information to solve/debug this problem, but this is
+      // not a problem in Raptor, so we do not want to add more specific logging here - this does
+      // however document that the problem exist. Turn on debug logging and move the start/end
+      // coordinate around until you see this message.
+      //
+      // See https://github.com/opentripplanner/OpenTripPlanner/issues/5601
+      LOG.warn(
```

Comment:
If have a reproduction case, I can take a look at it and start thinking about a solution.

Reply:
I did not want to make the fix in this PR - if you can do it I am very happy. Also, this is not causing any real problems - just very annoying. But, of cause if this error also result in dropped paths, then there is a problem. It is easy to reproduce, I think I described it in the issue.